tests/common/widgets/dynamic-widgets/options.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  createAlgoliaSearchClient,
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '@instantsearch/mocks';
+import { wait } from '@instantsearch/testutils';
+import { within, screen } from '@testing-library/dom';
+
+import type { DynamicWidgetsWidgetSetup } from '.';
+import type { TestOptions } from '../../common';
+
+export function createOptionsTests(
+  setup: DynamicWidgetsWidgetSetup,
+  { act }: Required<TestOptions>
+) {
+  describe('options', () => {
+    const searchClient = createAlgoliaSearchClient({
+      search: jest.fn((requests) => {
+        return Promise.resolve(
+          createMultiSearchResponse(
+            ...requests.map(() => {
+              return createSingleSearchResponse({
+                facets: {
+                  brand: {
+                    Samsung: 633,
+                    Metra: 591,
+                  },
+                  category: {
+                    TV: 633,
+                    Radio: 591,
+                  },
+                },
+                renderingContent: {
+                  facetOrdering: {
+                    facets: {
+                      order: ['category', 'brand'],
+                    },
+                  },
+                },
+              });
+            })
+          )
+        );
+      }),
+    });
+
+    beforeEach(() => {
+      searchClient.search.mockClear();
+    });
+
+    it('renders with default options', async () => {
+      await setup({
+        instantSearchOptions: {
+          searchClient,
+          indexName: 'indexName',
+          initialUiState: {
+            indexName: {},
+          },
+        },
+        widgetParams: {},
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      const dynamicWidgets = Array.from(
+        document.querySelector('.ais-DynamicWidgets')!.childNodes
+        // Vue 3 outputs comment nodes
+      ).filter((node) => node.nodeType === Node.ELEMENT_NODE);
+
+      expect(dynamicWidgets).toHaveLength(2);
+
+      expect(
+        within(dynamicWidgets[0] as HTMLElement).getByRole('link', {
+          name: /tv/i,
+        })
+      ).toBeInTheDocument();
+
+      expect(
+        within(dynamicWidgets[1] as HTMLElement).getByRole('checkbox', {
+          name: /samsung/i,
+        })
+      ).toBeInTheDocument();
+    });
```

Comment:
Could we test against an inline snapshot instead? It makes it easier to understand what the full output is.

Reply:
won't the output be huge?

Reply:
Yes, also we can't because the output is very different between vanilla and React as we don't wrap the widgets into a div in React